pkg/cvo/cvo_test.go

```diff
@@ -3272,190 +3272,97 @@ func fakeClientsetWithUpdates(obj *configv1.ClusterVersion) *fake.Clientset {
 	tests := []struct {
-		name          string
-		update        *payload.Update
-		want          bool
-		wantErr       bool
-		wantVerifiers int
+		name             string
+		fileName         string
+		update           *payload.Update
+		expectedError    string
+		expectedVerifier string
+		expectStore      bool
 	}{
 		{
-			name:   "is a no-op when no objects are found",
-			update: &payload.Update{},
+			name:     "is a no-op when no objects are found",
+			fileName: "",
+			update:   &payload.Update{},
 		},
 		{
-			name: "requires data",
-			update: &payload.Update{
-				Manifests: []lib.Manifest{
-					{
-						GVK: schema.GroupVersionKind{Version: "v1", Kind: "ConfigMap"},
-						Obj: &unstructured.Unstructured{
-							Object: map[string]interface{}{
-								"metadata": map[string]interface{}{
-									"name":      "release-verification",
-									"namespace": "openshift-config-managed",
-									"annotations": map[string]interface{}{
-										"release.openshift.io/verification-config-map": "",
-									},
-								},
-							},
-						},
-					},
-				},
-			},
-			wantErr: true,
+			name:          "requires data",
+			fileName:      "requires-data.yaml",
+			update:        &payload.Update{},
+			expectedError: ExpectedError,
 		},
 		{
-			name: "requires stores",
-			update: &payload.Update{
-				Manifests: []lib.Manifest{
-					{
-						GVK: schema.GroupVersionKind{Version: "v1", Kind: "ConfigMap"},
-						Obj: &unstructured.Unstructured{
-							Object: map[string]interface{}{
-								"metadata": map[string]interface{}{
-									"name":      "verification",
-									"namespace": "openshift-config",
-									"annotations": map[string]interface{}{
-										"release.openshift.io/verification-config-map": "",
-									},
-								},
-								"data": map[string]interface{}{
-									"verifier-public-key-redhat": string(redhatData),
-								},
-							},
-						},
-					},
-				},
-			},
-			wantErr: true,
+			name:          "requires stores",
+			fileName:      "requires-stores.yaml",
+			update:        &payload.Update{},
+			expectedError: ExpectedError,
 		},
 		{
-			name: "requires verifiers",
-			update: &payload.Update{
-				Manifests: []lib.Manifest{
-					{
-						GVK: schema.GroupVersionKind{Version: "v1", Kind: "ConfigMap"},
-						Obj: &unstructured.Unstructured{
-							Object: map[string]interface{}{
-								"metadata": map[string]interface{}{
-									"name":      "release-verification",
-									"namespace": "openshift-config-managed",
-									"annotations": map[string]interface{}{
-										"release.openshift.io/verification-config-map": "",
-									},
-								},
-								"data": map[string]interface{}{
-									"store-local": "file://../verify/testdata/signatures",
-								},
-							},
-						},
-					},
-				},
-			},
-			wantErr: true,
+			name:          "requires verifiers",
+			fileName:      "requires-verifiers.yaml",
+			update:        &payload.Update{},
+			expectedError: ExpectedError,
```

Comment:
Hmm, `requires-verifiers.yaml` declares `store-local`.  Why are we saying `...did not provide any signature stores...` for it?

Reply:
`expectStore` indicates whether we expect `loadConfigMapVerifierDataFromUpdate` to return non nil 'signatureStore' which it only does for the last two test cases where `expectStore` is set to `true`. This is per the original test cases as well but they used `want: true`.